menuxconfig.py

```diff
@@ -0,0 +1,765 @@
+
+        self.tree.tag_configure(
+            'evenn',
+            background='light goldenrod yellow',
+            foreground='wheat4')
+        self.tree.tag_configure(
+            'eveny',
+            background='light yellow',
+            foreground='green3')
+        self.tree.tag_configure(
+            'evenm',
+            background='light yellow',
+            foreground='green3')
+        self.tree.tag_configure(
+            'evenv',
+            background='light yellow',
+            foreground='blue')
+
+        self.tree.tag_configure('menu', background='bisque2')
+        self.tree.tag_configure(
+            'choice',
+            background='lavender',
+            foreground='DarkOrange4')
+        self.tree.tag_configure(
+            'comment',
+            background='light blue',
+            foreground='purple3')
+
+    def value_str(self, sc):
+
+        tags = self.tag_str(sc)
+        if sc.type in (STRING, INT, HEX):
+            return "{}v".format(tags), "{}".format(sc.str_value)
+
+        # BOOL or TRISTATE
+
+        # The choice mode is an upper bound on the visibility of choice symbols, so
+        # we can check the choice symbols' own visibility to see if the choice is
+        # in y mode
+        if isinstance(sc, Symbol) and sc.choice and sc.visibility == 2:
+            # For choices in y mode, print '<--' next to the selected symbol
+            if sc.choice.selection is sc:
+                return "{}y".format(tags), "<--"
+            else:
+                return "{}n".format(tags), "   "
+
+        tri_val_str = ("n", "M", "y")[sc.tri_value]
+        tri_val_tag = ("n", "m", "y")[sc.tri_value]
+
+        final_tag = tags + tri_val_tag
+        if len(sc.assignable) == 1:
+            # Pinned to a single value
+            return final_tag, "-{}-".format(tri_val_str)
+
+        if sc.type == BOOL:
+            return final_tag, "[{}]".format(tri_val_str)
+
+        if sc.type == TRISTATE:
+            if sc.assignable == (1, 2):
+                # m and y available
+                # Gets a bit confusing with .format()
+                return final_tag, "{" + tri_val_str + "}"
+            return final_tag, "<{}>".format(tri_val_str)
+
+    def tag_str(self, sc):
+        tag = 'odd' if self.odd else 'even'
+        self.odd = not self.odd
+
+        if sc == MENU:
+            return 'menu'
+        if sc == COMMENT:
+            return 'comment'
+
+        if isinstance(sc, Symbol):
+            return 'odd' if self.odd else 'even'
+
+        if isinstance(sc, Choice):
+            return 'choice'
+
+        return ''
+
+    def type_str(self, sc):
+        if sc.type == BOOL:
+            return 'bool'
+        if sc.type == TRISTATE:
+            return 'tristate'
+        if sc.type is HEX:
+            return 'hex'
+        if sc.type is INT:
+            return 'integer'
+        if sc.type is STRING:
+            return 'string'
+        return 'unknown type'
```

Comment:
Could use `Kconfiglib.TYPE_TO_STR` here. If you want to use a special string for `UNKNOWN`, you could do this:

```python
if sc.type == UNKNOWN:
    return 'unknown type'
return TYPE_TO_STR[sc.type]
```